### PR TITLE
Update authoring_commands.md

### PR DESCRIPTION
**Pull request by tjprescott**
Adds guidance for suppressing `--ids` on list commands and supporting "name or id"-type parameters.

### CODE REVIEW

**Comment by promptws** [comment]
View a preview at https://prompt.ws/r/Azure/azure-cli/5838
This is an **experimental preview** for @microsoft users.

**Comment by derekbekoe** [comment]
@tjprescott Can we get https://github.com/Azure/azure-cli/pull/5769 in then this?

**Comment by tjprescott** [comment]
@derekbekoe makes sense to me.

File: doc/authoring_command_modules/authoring_commands.md [hunk 1]
```diff
@@ -18,20 +18,24 @@ The document provides instructions and guidelines on how to author individual co
...
+
+[12. Validators](#validators)
 
-[12. Registering Flag Arguments](#registering-flags)
 
 [13. Registering Enum Arguments](#registering-enums)
 
 [14. Preventing particular extensions from being loading](#extension-suppression)
 
+[15. Preventing particular extensions from being loading](#extension-suppression)
```
Comment (derekbekoe) [inline doc/authoring_command_modules/authoring_commands.md]: This is duplicated.

Comment (tjprescott) [reply doc/authoring_command_modules/authoring_commands.md]: Yeah, I had to commit the merge and then fix it up.  Should be good now.

**Comment by tjprescott** [comment]
Merging since this is readme only and thus pylint is irrelevant and being fixed by a separate PR.

**Pull request merged**